go.mod

```diff
@@ -4,28 +4,28 @@ go 1.16
 
 require (
 	github.com/agext/levenshtein v1.2.3 // indirect
+	github.com/cenkalti/backoff/v4 v4.1.0
```

Comment:
I wonder if you'd be able to use the exponential backoff that the kubernetes API uses. https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L48

Reply:
Oh, I missed this link and wasn't aware of that mechanism. I added retries with 3rd party exp. back-off package. I'll switch to using this one instead.